youtube.go

```diff
@@ -432,7 +413,9 @@ func (y *Youtube) getVideoInfo() error {
 	if err != nil {
 		return err
 	}
-	defer resp.Body.Close()
+	defer func() {
+		err = resp.Body.Close()
```

Comment:
mistake,  if err not nil, here will overwrite existing err

Reply:
I was wondering if we can adopt Thanos [runutil](https://thanos.io/contributing/coding-style-guide.md/#exhaust-readers) to handle this kind of error

Reply:
We can,  but  I found ExhaustCloseWithLogOnErr can be implemented without this pkg, so I write a custom close method.